### DISCUSSION TITLE
Any way to go back into history of a file?

### DISCUSSION THREAD

**Discussion by gnohj** [LazyVim/LazyVim#4647] [Q&A]
For ex in vscode, there's a previous icon that appears in the upper right toolbar for every buffer. When I click on it- it opens a new file and shows a diff of the current/prev commit.. and if I wanted to compare that commit to its previous, I can just click the icon again for that new file. Is there a plugin/keymap that provides this same functionality?

**Comment by rogeriofrsouza** [comment]
There's `Lazygit Current File History`, which is activated with the keymap `<leader>gf`

**Reply by alewis001** [reply]
That's not quite the case in the latest lazyvim. The `<leader>gf` will open a Snacks picker to go through the history (shows the file commit history and a preview pane of the diff). If you hit Enter on a commit, it will checkout that commit.

If you specifically want Lazygit based file history, which is what I wanted, I added the following to my keymaps.lua

```
vim.keymap.set("n", "<leader>gf", function(opts)
  Snacks.lazygit.log_file(opts)
end, { desc = "Lazygit current file history" })
```
